Added DeepSTARR

bernardo-de-almeida opened the pull request:
Add the DeepSTARR model for DNA sequence prediction of enhancer activities.

Zenodo: https://doi.org/10.5281/zenodo.5502060
Code: https://github.com/bernardo-de-almeida/DeepSTARR/tree/main/DeepSTARR
Manuscript: https://www.biorxiv.org/content/10.1101/2021.10.05.463203

haimasree commented on DeepSTARR/model.yaml:
```diff
@@ -0,0 +1,46 @@
+
+schema:
+    inputs:
+        shape: (249,4)
```
I took a quick look into the failing tests. I think changing ```shape: (249, 4)``` to ```shape: (249, 4, 1)``` will fix the test

bernardo-de-almeida replied:
I changed the shape as suggested, although the input shape in the model is (249, 4). See if it fixes the test

haimasree commented on DeepSTARR/model.yaml:
```diff
@@ -0,0 +1,46 @@
+dependencies:
+    conda:
+      - python=3.7
+      - h5py
```
This does not matter for now but for ensuring that the model continues to operate as expected in future it is important to pin the dependencies. 
I am providing the versions as resolved in CI. Please consider adding them here.
```- h5py=3.6.0```

haimasree commented on DeepSTARR/model.yaml:
```diff
@@ -0,0 +1,46 @@
+    conda:
+      - python=3.7
+      - h5py
+      - pip
```
```- pip=21.2.2```

haimasree commented on DeepSTARR/model.yaml:
```diff
@@ -0,0 +1,46 @@
+      - h5py
+      - pip
+    pip:
+      - keras>=2.0.4
```
```- keras==2.7.0```

haimasree commented on DeepSTARR/model.yaml:
```diff
@@ -0,0 +1,46 @@
+      - pip
+    pip:
+      - keras>=2.0.4
+      - tensorflow>=1.0
```
```- tensorflow==2.7.0

haimasree reviewed (changes requested):
Thanks for the pr. I have provided (untested) a fix for the failing test and have provided a few more suggestions.I will accept the pr as soon as the tests are fixed.

bernardo-de-almeida commented:
I have performed the suggested changes, can you test it now? Many thanks

bernardo-de-almeida commented:
We have CentOS 7.9 and pyrle 0.0.33.
Before I added the versions I currently use in my environment, but I updated now for your suggested pinned versions since they were from a successfully resolved conda environment

haimasree commented:
> We have CentOS 7.9 and pyrle 0.0.33. Before I added the versions I currently use in my environment, but I updated now for your suggested pinned versions since they were from a successfully resolved conda environment

Great! Thanks. The main test is passing now :). The other test is failing due to a bug in kipoi. I am working on a fix as we speak. I will merge this pr afterwards.

bernardo-de-almeida commented:
Great, many thanks for your help!
